include/aws/s3/private/s3_copy_object.h

```diff
@@ -32,7 +32,7 @@ struct aws_s3_copy_object {
 
     /* Members to only be used when the mutex in the base type is locked. */
     struct {
-        struct aws_array_list etag_list;
```

Comment:
nit: does it actually make sense to use aws_s3_put_part_info here(comment about list actual type would also be nice)?
copy for example would never need checksums. 
should name at least be more generic?

Reply:
[UploadPartCopy](https://docs.aws.amazon.com/AmazonS3/latest/API/API_UploadPartCopy.html#API_UploadPartCopy_ResponseSyntax) CAN actually generate a checksum, and we could use them some day...

The reason I changed s3_copy_object.h/c to use the new `aws_s3_put_part_info` is that it also needs to create a CompleteMultipartUpload request, and I changed the function that creates it to take a "part list" instead of separate "etag list" and "checksum list". So both `s3_copy_object.h` and `s3_auto_ranged_put.h` need to agree on how they're storing part info. I didn't realize we needed this in copy ... so agree the name isn't super generic. But basically it's "the data you'll need for CompleteMultipartUpload later"

The reason the name isn't super generic is that I started with `aws_s3_part_info` and discovered that name was already being used by the ListParts call. I could just rename that one to `aws_s3_list_part_info` and steal the nice short generic name for use here, or think of a more generic name like `aws_s3_upload_part_info`??? `aws_s3_multipart_info`???

Reply:
aws_s3_mpu_part_info !
good name
thanks